sensors/hall effect sensor/rpm/hall_effect_sensor.h

```diff
@@ -0,0 +1,46 @@
+//NOTE: CODE TAKEN FROM: https://www.youtube.com/watch?v=u2uJMJWsfsg
+
+#ifndef HALL_EFFECT_SENSOR_H
+#define HALL_EFFECT_SENSOR_H
+
+class HallEffectSensor : public Sensor
```

Comment:
If this is a child class of base_sensor's Sensor class, you'll need to include base_sensor.h